File: dotnet/u02-unit-testing/it/unibo/testlecture/u02_unit/DeviceImpl.cs
using System;

namespace it.unibo.testlecture.u02_unit_testing
{
    public class DeviceImpl : Device
    {
        public bool isOn { get; set; } 
        
    	public DeviceImpl(){
    	    this.isOn = false;  
    	}
    	
        public void on() { isOn = true; }
        public void off() { isOn = false; }
    }
}

File: jvm/src/main/java/it/unibo/testlecture/u04_tdd/DeviceImpl.java
package it.unibo.testlecture.u04_tdd;

public class DeviceImpl implements Device {
    private Boolean state;

    public DeviceImpl(){
        this.state = false;
    }

    @Override
    public void on() {
        this.state = true;
    }

    @Override
    public void off() {
        this.state = false;
    }

    @Override
    public Boolean isOn() {
        return state;
    }
}
